code-studio/src/js/components/GridEditor.jsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * @overview React component to allow for easy editing of Karel Grids.
+ * Used in LevelBuilder, and relies on some apps code for validation.
+ * Supports both Bee and Farmer skins.
+ */
+/* global React, dashboard */
+
+var BeeCell = require('@cdo/apps/maze/beeCell');
+var Cell = require('@cdo/apps/maze/cell');
+var mazeUtils = require('@cdo/apps/maze/mazeUtils');
+
+var BeeCellEditor = require('./BeeCellEditor.jsx');
+var CellEditor = require('./CellEditor.jsx');
+var Grid = require('./Grid.jsx');
+
+window.dashboard = window.dashboard || {};
+
+var CellJSON = React.createClass({
+  propTypes: {
+    serialization: React.PropTypes.arrayOf(React.PropTypes.arrayOf(React.PropTypes.object)),
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  componentDidUpdate: function () {
+    var node = this.refs.serializedInput.getDOMNode();
+    node.focus();
+    node.select();
+  },
+
+  onChange: function (event) {
+    this.props.onChange(JSON.parse(event.target.value));
+  },
+
+  render: function () {
+    return (<label>
+      Cell JSON (for copy/pasting):
+      <input type="text" value={JSON.stringify(this.props.serialization)} ref="serializedInput" onChange={this.onChange}/>
+    </label>);
+  }
+});
+
+window.dashboard.GridEditor = React.createClass({
```

Comment:
I would recommend exporting this component directly as well as sticking it on dashboard, so that callers can also do
`var GridEditor = require('path/to/GridEditor.jsx')`
i.e.
```
window.dashboard.GridEditor = module.exports = React.createClass({...
```